src/notification/email_sender.py

```diff
@@ -169,25 +169,36 @@ def convert_report_to_dataframe(self) -> pd.DataFrame:
             "Data",
         ]
         del_header = True
-        del_single_group = False
-        del_single_department = False
+        del_single_group = True
+        del_single_department = True
 
         for search in self.search_report:
             if search["header"] is not None:
                 del_header = False
-            if "single_group" in search["result"]:
-                del_single_group = True
-            for _, terms in search["result"].items():
-                for _, departments_data in terms.items():
-                    if "single_department" in departments_data:
-                        del_single_department = True
 
+            for group, search_result in search["result"].items():
+                if group is not "single_group":
+                    del_single_group = False
+                for _, term_results in search_result.items():
+                    for dpt,_ in term_results.items():
+                        if dpt is not "single_department":
```

Comment:
Para comparar varíavel e string, usar `==` em vez de `is`. `is` é para comparar se o objeto é o mesmo objeto, não para comparar valores (ex.: `var is None`, já que em Python `None` é um singleton, um único objeto)